src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -704,8 +704,8 @@ public sealed interface MemorySegment permits AbstractMemorySegmentImpl {
     MemorySegment asSlice(long offset);
 
     /**
-     * Returns a new memory segment that has the same address and scope as this segment,
-     * but with the provided size.
+     * Returns a new memory segment that has the same address, scope, and
+     * read-only state as this segment, but with the provided size.
```

Comment:
In other parts of the javadoc we say something like this:

```
* If the buffer is {@linkplain Buffer#isReadOnly() read-only}, the resulting segment
     * is also {@linkplain ByteBuffer#isReadOnly() read-only}
```

I think this is better, as read-only is a bit of info that is not IMHO worth spelling out in the method javadoc first para,

Also, the term "read-only state" does not appear anywhere else in the FFM javadoc, so I'd recommend against using it. We can just say "if this segment is (link) read-only, the returned segment is read-only", or something like that.